components/channel_header_mobile/channel_info_button/index.ts

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+import {ActionCreatorsMapObject, bindActionCreators, Dispatch} from 'redux';
+
+import {ActionFunc, GenericAction} from 'mattermost-redux/types/actions';
+
+import {openModal} from 'actions/views/modals';
+
+import ChannelInfoButton from './channel_info_button';
+
+type Actions = {
+    openModal: (modalData: {modalId: string; dialogType: React.Component; dialogProps?: any}) => Promise<{
+        data: boolean;
+    }>;
+}
```

Comment:
One option could be to use `ComponentProps<T>` to extract the types from the component itself. 0/5
```tsx
type Actions = ComponentProps<typeof ChannelInfoButton>['actions'];
```